File: src/lib/software/audacity.py
from PySide6.QtCore import Slot
from src.lib.software import BaseSoftware, SoftwareCategory
from src.lib.github_release_scraper import GithubReleaseScraper

class Audacity(BaseSoftware):
    def __init__(self):
        super().__init__()

        self._gh = GithubReleaseScraper('audacity', 'audacity', self)
        self._gh.releases_scraped.connect(self._on_releases_scraped)

        self.key = 'audacity'
        self.name = 'Audacity'
        self.category = SoftwareCategory.Media
        self.download_name = 'audacity-win-64bit.exe'
        self.is_archive = False
        self.should_cache_url = True
        self.requires_admin = False
        self.icon = 'audacity.png'
        self.homepage = 'https://audacityteam.org'

    @Slot(list)
    def _on_releases_scraped(self, releases: list[str]):
        asset = next((release for release in releases if release.endswith('-64bit.exe')), None)
        if asset:
            self.url_resolved.emit(asset)
        else:
            self.error_occurred.emit(self.ResolveError.GitHubAssetNotFoundError)

    def resolve_download_url(self):
        self._gh.get_repo_releases()

File: packages/app/src/services/software/definitions/audacity.ts
import { ok, err } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import { GithubReleasesScraper } from '../githubReleasesScraper';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'audacity';
	public name = 'Audacity';
	public category = [SoftwareCategory.Audio, SoftwareCategory.Media];
	public downloadName = 'audacity-win-64bit.exe';
	public shouldCacheUrl = true;
	public icon = 'audacity.png';
	public homepage = 'https://audacityteam.org';

	public async resolveDownloadUrl() {
		const owner   = 'audacity';
		const repo    = 'audacity';
		const release = await GithubReleasesScraper.getRelease(owner, repo, r => r.endsWith('-64bit.exe'));
		if (release.isErr()) {
			return err(release.error);
		}

		return ok(release.value);
	}
}
